### ISSUE TITLE
project function of map is called twice

### ISSUE THREAD

**Issue by maiermic** [staltz/xstream#98]
The bug https://github.com/cyclejs/todomvc-cycle/issues/35 in the TODO-MVC example of Cycles.js is introduced by `xstream@5.2.2` and is still present in the current latest release `xstream@5.3.2`. As far as I can tell, the project function of map (returned by `amendStateWithChildren(sources.DOM)`) in [TaskList/index.js#L66](https://github.com/cyclejs/todomvc-cycle/blob/master/src/components/TaskList/index.js#L66) is called twice for each event. You can check this assumption by adding a debug statement in line 66

```js
let amendedState$ = state$.debug('state')
    .map(amendStateWithChildren(sources.DOM))
    .remember();
```

and a log statement between line 14 and 15:

```js
function amendStateWithChildren(DOMSource) {
  return function (todosData) {
    console.log('amendStateWithChildren', todosData);
    return {
      ...
```

If you run the app you see that each time you add an item to the list the console logs

```
state: Object {list: Array[1], filter: ""}
amendStateWithChildren Object {list: Array[1], filter: ""}
amendStateWithChildren Object {list: Array[1], filter: ""}
```

or something similar. Like I said: For each event of `state$`, the project function passed to `state$.map` is called twice.

**Comment by staltz** [comment]
Interesting bug. It's definitely a bug, and definitely related to `remember`. But here's one interesting bit: 
```diff
  return sourceTodosData$.map(sourceTodosData =>
    reducer$.fold((todosData, reducer) => reducer(todosData), sourceTodosData)
  ).flatten()
- .remember();
```
Removing that (in model()) fixes the issue, because then the `debug('state')` will also emit. The double emission is expected, it's app logic. The first emission is due to the fold seed, the second is due to changeRouteReducer$'s startWith.

Still, we need to fix remember() and MemoryStream.

**Comment by staltz** [comment]
Fixed in v7.0
